tools/text_processing_deployment/export_grammars.sh

```diff
@@ -61,23 +63,40 @@ echo "FORCE_REBUILD = $FORCE_REBUILD"
 echo "WHITELIST = $WHITELIST"
 
 
-if [[ ${OVERWRITE_CACHE,,} == "true" ]]; then
+if [[ ${OVERWRITE_CACHE,,} == "true" ]] ; then
   OVERWRITE_CACHE="--overwrite_cache "
-  python3 pynini_export.py --output_dir=. --grammars=${GRAMMARS} --input_case=${INPUT_CASE} \
-    --language=${LANGUAGE} --cache_dir=${CACHE_DIR} --whitelist=${WHITELIST} ${OVERWRITE_CACHE} || exit 1
   else OVERWRITE_CACHE=""
 fi
 
+CLASSIFY_FAR=${CACHE_DIR}"/classify/tokenize_and_classify.far"
+VERBALIZE_FAR=${CACHE_DIR}"/verbalize/verbalize.far"
+
+if [[ -f $CLASSIFY_FAR ]] && [[ -f $VERBALIZE_FAR ]] && [[ ${OVERWRITE_CACHE} == "" ]]; then
+  SKIP_FAR_CREATION="True"
+  echo "Far files exists and OVERWRITE_CACHE is set to False"
+fi
+
+if [[ ${SKIP_FAR_CREATION} != "True" ]]; then
+  python3 pynini_export.py --output_dir=${FAR_PATH} --grammars=${GRAMMARS} --input_case=${INPUT_CASE} \
+    --language=${LANGUAGE} --cache_dir=${CACHE_DIR} --whitelist=${WHITELIST} ${OVERWRITE_CACHE} || exit 1
+fi
+
 if [[ ${FORCE_REBUILD,,} == "true" ]]; then
   FORCE_REBUILD="--no-cache"
   else FORCE_REBUILD=""
 fi
 
 find . -name "Makefile" -type f -delete
-bash docker/build.sh $FORCE_REBUILD
+
+
+
+
 
 if [[ ${MODE} == "test" ]]; then
   MODE=${MODE}_${GRAMMARS}
+  bash docker/build.sh $FORCE_REBUILD
+else
+  exit 0
```

Comment:
Could this be revert to have a way to interactively launch the docker? i.e. MODE!="test" => export grammars + launch docker to interactively try out grammars

Reply:
added `interactive` option